src/client.rs

```diff
@@ -502,6 +534,38 @@ impl Client {
         authentication: &RegistryAuth,
         operation: RegistryOperation,
     ) -> Result<Option<String>> {
+        self.store_auth_if_needed(image.resolve_registry(), authentication)
+            .await;
+        // preserve old caching behavior
```

Comment:
Do we actually need to preserve the old behavior? I think a call to `get_auth_token` should do the same logic of fetching the token and doing the auth step, right?

Reply:
I didn't want to break API, and that could be that user wants to do auth and it might not really call auth underneath (because cache) feels a bit vile.

Reply:
Ok, that makes sense